### ISSUE TITLE
Updated analysis: HGAT subtyping - modify gliomatosis cerebri inclusion

### ISSUE THREAD

**Issue by jharenza** [AlexsLemonade/OpenPBTA-analysis#972]
#### What analysis module should be updated and why?
HGG subtyping - upon inspection of `pathology_free_text_diagnosis` and discussion with Cassie Kline, I noticed that only 2/4 gliomatosis cerebri samples we subtyped are denoted as high grade (`anaplastic gliomatosis cerebri (who grade 4)`), therefore, only those two should be subtyped within the HGG subtyping module. 

```
  Kids_First_Biospecimen_ID sample_id pathology_diagnosis                pathology_free_text_diagnosis
1               BS_F8K4VQMF 7316-3817 Gliomatosis Cerebri                          gliomatosis cerebri
2               BS_GG9W3J9Y 7316-3817 Gliomatosis Cerebri                          gliomatosis cerebri
3               BS_MSB1RDAJ   7316-71 Gliomatosis Cerebri anaplastic gliomatosis cerebri (who grade 4)
4               BS_X4X1SFSE   7316-71 Gliomatosis Cerebri anaplastic gliomatosis cerebri (who grade 4)
```

#### What changes need to be made? Please provide enough detail for another participant to make the update.
Since gliomatosis can occur in HGG or LGG context, we should update the module to pull only those which have a high grade (3 or 4) designation for this module. If no grade, then we should not subtype.

As a sanity check, I did check the path report for 7316-3817 and found that actually, this `pathology_diagnosis` was either mis-entered into our BRP (path report states HGG, GBM grade IV) or it was updated to gliomatosis later. Will have our team check on this and if needed, update for the v19 histology file release. In any case, it will likely be subtyped with this module, but best to add this caveat as well.

#### What input data should be used? Which data were used in the version being updated?
same as previous/v18

#### When do you expect the revised analysis will be completed?
this week

#### Who will complete the updated analysis?
@kgaonkar6 or @jharenza

**Comment by kgaonkar6** [comment]
Just wanted to check this issue is still something we want to work on since 7316-3817 is now updated to glioblastoma?

**Comment by jharenza** [comment]
Yes please, just because it can be high or low grade and we don't want any low grades sneaking in. We'll get the same result, but when we add more samples in the future, this won't be a worry. Thanks!

**Comment by kgaonkar6** [comment]
Ok cool, since pathology_free_text_diagnosis is so varied in it's terms I wanted to confirm should I make an exclusion to remove if pathology_free_text_diagnosis exactly matches `gliomatosis cerebri` ?

Since pathology_free_text_dx is not quite uniform the grade is sometimes written as `grade III` or `grade IV` or `grade 3` or `grade 4` or sometimes as `grade iii/iv` or `grade 3-4` or `gr 3`  or `gr 4` across the path_free_text_dx so I was thinking a simple exclusion with pathology_diagnosis == "Gliomatosis Cerebri" and pathology_free_text_diagnosis ==`gliomatosis cerebri` would work, thoughts?

**Comment by jharenza** [comment]
For this case, we would only include in HGG subtyping if there is any annotation in HGG denoting the grade - I think for this, we should just match the exact terms in the JSON file as we did with embryonal tumor subtyping [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-embryonal/subset-files/embryonal_subtyping_path_dx_strings.json). So, IF `pathology_diagnosis == Gliomatosis Cerebri` AND `include_free_text == anaplastic gliomatosis cerebri (who grade 4)` for this case. If we get more, we can add on, but if we see `pathology_diagnosis == Gliomatosis Cerebri` with no grading in free text, these will not be subtyped in HGG, which is what we want. Does that make sense?

**Comment by kgaonkar6** [comment]
got it, makes sense!

**Comment by kgaonkar6** [comment]
fixed with #982
